packages/manager/core/application/src/ovh-tracking.tsx

```diff
@@ -0,0 +1,146 @@
+      const name = `${universe}::${applicationName}::${partsReplace ||
+        'homepage'}`;
+      tracking.trackPage({
+        name,
+        level2: trackLevel2(universe),
+        page_category: path || 'homepage',
+      });
+    });
+  };
+
+  useEffect(() => {
+    tracking.init(false);
+  }, []);
+
+  useEffect(() => {
+    // When location change, send tracking load page
+    OvhTrackPage(location);
+    myStateRef.current = location;
+    setLocationPath(location);
+  }, [location]);
+
+  const ovhTrackingSendCLick = (value: string) => {
+    const path = myStateRef.current.pathname.split('/')[1];
+    env.then((response) => {
+      const { applicationName, universe } = response;
+      const name = `${applicationName}::${path || 'homepage'}::${value}`;
+      const page = `${universe}::${applicationName}::${path || 'homepage'}`;
+      tracking.trackClick({
+        name,
+        page: { name: page },
+        complete_page_name: page,
+        level2: trackLevel2(universe),
+        type: 'action',
+      });
+    });
+  };
+
+  // Send tracking click for the ODS_COMPONENTS_GROUP
+  const ovhTrackingParentNode = (event) => {
+    const value = event.detail.value || event.detail.newValue;
+    ovhTrackingSendCLick(value);
+  };
+
+  // Init event listener for each ODS_COMPONENTS_GROUP on value change
+  const addOdsEventListenerValue = () => {
+    const elmTab = [];
+    ODS_COMPONENTS_GROUP.forEach((element) => {
+      const odsSelector = document.querySelectorAll(element);
+      odsSelector.forEach((elementSlctr) => {
+        elementSlctr?.addEventListener(
+          'odsValueChange',
+          ovhTrackingParentNode,
+          false,
+        );
+        elmTab.push(elementSlctr);
+      });
+    });
+    return elmTab;
+  };
+
+  useLayoutEffect(() => {
+    let tabsEvents = [];
+    const timer = setTimeout(() => {
+      // After the layout and all ods components loaded
+      // Add events listener
+      tabsEvents = addOdsEventListenerValue();
+    }, 500);
+
+    // Clean every listener events
+    return () => {
+      clearTimeout(timer);
+      if (tabsEvents.length > 0)
+        tabsEvents.forEach((element) => {
+          element.removeEventListener(
+            'odsValueChange',
+            ovhTrackingParentNode,
+            false,
+          );
+        });
+    };
+  }, [location]);
+
+  // Get the data-tracking value from ODS_COMPONENTS
+  // To send a track click
+  const ohvTrackingAction = (event) => {
+    const element = event.target as HTMLElement;
+    if (
+      ODS_COMPONENTS.indexOf(element.tagName) > -1 &&
+      element['data-tracking']
+    )
+      ovhTrackingSendCLick(element['data-tracking']);
+  };
+
+  useEffect(() => {
+    // Add event listener click on the window
+    window.addEventListener('click', ohvTrackingAction, false);
+    return () => {
+      // clean all events listener
+      window.removeEventListener('click', ohvTrackingAction, false);
+    };
+  }, []);
```

Comment:
**question:** Scanning the entire window object on every user click can cause performance issues no?